outbound-dialer/src/main/java/org/motechproject/nms/outbounddialer/service/impl/TargetFileServiceImpl.java

```diff
@@ -0,0 +1,287 @@
+                    writer.write(",");
+                    writer.write(callRetry.getLanguageLocationCode());
+                    //todo...
+                    writer.write("\n");
+                }
+
+                page++;
+                recordCount += numBlockRecord;
+
+            } while (numBlockRecord > 0);
+
+            LOGGER.info("Created targetFile with {} record{}", recordCount, recordCount == 1 ? "" : "s");
+
+        } catch (NoSuchAlgorithmException | IOException e) {
+            LOGGER.error(e.getMessage());
+            alertService.create(targetFile.toString(), "targetFile", e.getMessage(), AlertType.CRITICAL,
+                    AlertStatus.NEW, 0, null);
+            return null;
+        }
+
+        String md5Checksum = new String(Hex.encodeHex(md.digest()));
+        TargetFileNotification tfn = new TargetFileNotification(targetFileName, md5Checksum, recordCount);
+        LOGGER.info("TargetFileNotification = {}", tfn.toString());
+        return tfn;
+    }
+
+
+    private void sendNotificationRequest(TargetFileNotification tfn) {
+        String notificationUrl = settingsFacade.getProperty(TARGET_FILE_NOTIFICATION_URL);
+        LOGGER.info("Sending {} to {}", tfn, notificationUrl);
+
+        try {
+            HttpClient httpClient = HttpClients.createDefault();
+            HttpPost httpPost = new HttpPost(notificationUrl);
+            ObjectMapper mapper = new ObjectMapper();
+            String requestJson = mapper.writeValueAsString(tfn);
+            httpPost.setHeader("Content-type", "application/json");
+            httpPost.setEntity(new StringEntity(requestJson));
+            HttpResponse response = httpClient.execute(httpPost);
+            int responseCode = response.getStatusLine().getStatusCode();
+            if (responseCode != HttpStatus.SC_OK) {
+                String error = String.format("Expecting HTTP 200 response from %s but received HTTP %d : %s ",
+                        notificationUrl, responseCode, EntityUtils.toString(response.getEntity()));
+                LOGGER.error(error);
+                alertService.create("targetFile notification request", "targetFile", error, AlertType.CRITICAL,
+                        AlertStatus.NEW, 0, null);
+            }
+        } catch (IOException e) {
+            LOGGER.error("Unable to send targetFile notification request: {}", e.getMessage());
+            alertService.create("targetFile notification request", "targetFile", e.getMessage(), AlertType.CRITICAL,
+                    AlertStatus.NEW, 0, null);
+        }
+    }
+
+
+    @MotechListener(subjects = { GENERATE_TARGET_FILE_EVENT })
+    public void generateTargetFile(MotechEvent event) {
+        LOGGER.info(event.toString());
+
+        TargetFileNotification tfn = generateTargetFile();
+
+        if (tfn != null) {
+            //notify the IVR system the file is ready
+            sendNotificationRequest(tfn);
+        }
```

Comment:
when tfn == null, we raise alerts and let this fall through, correct? Is there a backup/catchup mechanism or are the calls for the day given up?